infrastructure/src/main/java/org/corfudb/infrastructure/RemoteMonitoringService.java

```diff
@@ -563,8 +563,14 @@ private DetectorTask detectFailure(PollReport pollReport) {
             return DETECTOR_TASK_SKIPPED;
         }
 
+        ClusterState clusterState = clusterContext.getClusterView();
+        if (clusterState.isPrimarySequencerReady()) {
```

Comment:
Would it be possible to bring it before L544 to avoid other checks as well?

Reply:
The reason I left it here is because this check examines the cluster state map which I thought may be more expensive compared to the other checks which appear to be cheaper in execution time. Please correct me if I am mistaken? thanks